maptivist/src/main/webapp/script.js

```diff
@@ -12,27 +12,149 @@
+    title: "Justice for George Floyd"
+  });
+
+  // Adds the new marker and infowindow to the map
+  marker.setMap(map);
+  var infowindow = new google.maps.InfoWindow({
+    content: '<button>This works<button>'
+  });
+  
+  marker.addListener('click', function() {
+    infowindow.open(map, marker);
+  });
+}
+
 /**
- * Adds a random greeting to the page.
+* Adds an Auto complete search box to the main map
  */
+function searchBox(){
+  // Create the search box, info windows for results and link them to their UI elements
+  var box = document.getElementById("search-container");
+  var input = document.getElementById("search-input");
+  var searchBox = new google.maps.places.SearchBox(input);
+  var infowindow = new google.maps.InfoWindow();
+  var infowindowContent = document.getElementById('infowindow-content');
+  infowindow.setContent(infowindowContent);
 
+  // Add to the map
+  map.controls[google.maps.ControlPosition.TOP_LEFT].push(box);
+  
+  // Set the data fields to return when the user selects a place.
+  var autocomplete = new google.maps.places.Autocomplete(input);
+  autocomplete.setFields(
+    ['address_components', 'geometry', 'icon', 'name']);
 
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+  autocomplete.addListener('place_changed', function() {
+    var place = autocomplete.getPlace();
+    if (!place.geometry) {
+        // User entered the name of a Place that was not suggested and
+        // pressed the Enter key, or the Place Details request failed.
+        window.alert("No details available for input: '" + place.name + "'");
+        return;
+        }
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    // If the place has a geometry, then present it on a map.
+    if (place.geometry.viewport) {
+        map.fitBounds(place.geometry.viewport);
+    } else {
+        map.setCenter(place.geometry.location);
+        map.setZoom(17);  
+    }
+    
+    // Concatentates a readable address for an autocomplete result
+    var address = '';
+        if (place.address_components) {
+            address = [
+                (place.address_components[0] && place.address_components[0].short_name || ''),
+                (place.address_components[1] && place.address_components[1].short_name || ''),
+                (place.address_components[2] && place.address_components[2].short_name || '')
+            ].join(' ');
+        }
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+        // Add the result contents to the infowindow display
+        infowindowContent.children['place-icon'].src = place.icon;
+        infowindowContent.children['place-name'].textContent = place.name;
+        infowindowContent.children['place-address'].textContent = address;
+        infowindow.open(map);
```

Comment:
The indentation looks wrong here compared to the previous line.

Reply:
Yeah I think that's a merge conflict because that shouldn't be there